src/mongo/db/storage/mmap_v1/mmap_windows.cpp

```diff
@@ -184,8 +184,8 @@ void* MemoryMappedFile::map(const char* filenameIn, unsigned long long& length)
     strncpy(filename, filenameIn, 255);
     filename[255] = 0;
     {
-        size_t len = strlen(filename);
-        for (size_t i = len - 1; i >= 0; i--) {
+        int len = strlen(filename);
```

Comment:
If len cannot be zero, it can have size_t type. But if type of iterator 'i' is size_t, condition 'i >= 0' is always true and it either is redundant or access to array at index ((size_t) -1) is possible. And endless cycle which ends with SIGSEGV is also possible.